syntax = "proto3";

package ws_backend;

service WS_Backend_Service {

  rpc SendTimestamp(SendTimestampRequest) returns (SendTimestampResponse) {}

  rpc RemoveAccess(RemoveAccessRequest) returns (RemoveAccessResponse) {}

  rpc GetUpdates(GetUpdatesRequest) returns (GetUpdatesResponse) {}

  rpc HandleDelete(HandleDeleteRequest) returns (HandleDeleteResponse) {}

}

message SendTimestampRequest {
  string document_id = 1;
  string timestamp = 2;
}

message SendTimestampResponse {
  string status = 1;
}

message RemoveAccessRequest {
  string user_id = 1;
  string document_id = 2;
}

message RemoveAccessResponse {
  string status = 1;
}

message GetUpdatesRequest {

}

message GetUpdatesResponse {
  message DocUpdates {
    string document_id = 1;
    repeated string update = 2;
  }
  repeated DocUpdates res = 3;
}

message HandleDeleteRequest {
  string document_id = 1;
}

message HandleDeleteResponse {
  string status = 1;
}
